### PR TITLE
test: Add tests for masking of sensitive data

### PR DESCRIPTION
**Pull request by mydea**
This tests masking of initial & updated values for password & credit card fields.

You can see that for now credit card data is not masked, which demonstrates the current behavior. 
In a future PR we can then verify that our fixed masking is working.

ref https://github.com/getsentry/sentry-javascript/issues/10258

### CODE REVIEW

File: packages/rrweb/test/__snapshots__/integration.test.ts.snap [hunk 1]
```diff
@@ -7236,7 +7236,7 @@ exports[`record integration tests mutations should work when blocked class is un
                         \\"attributes\\": {
                           \\"class\\": \\"rr-block\\",
                           \\"rr_width\\": \\"1904px\\",
-                          \\"rr_height\\": \\"21px\\"
+                          \\"rr_height\\": \\"21.5px\\"
```
Comment (billyvg) [inline packages/rrweb/test/__snapshots__/integration.test.ts.snap:7239]: This is flakey :(

File: packages/rrweb-snapshot/test/html/form-fields-sensitive.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      </label>
+      <label>
+        <input autocomplete="cc-exp-month" value="initial" />
+      </label>
+      <label>
+        <input autocomplete="cc-exp-year" value="initial" />
+      </label>
+      <label>
+        <input autocomplete="cc-csc" value="initial" />
+      </label>
```
Comment (billyvg) [inline packages/rrweb-snapshot/test/html/form-fields-sensitive.html:35]: LGTM

(ref: https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fe-autocomplete-cc-name)

**Review by billyvg — APPROVED** [review]

File: packages/rrweb/test/__snapshots__/integration.test.ts.snap (hunk 1 above)
Comment (mydea) [reply packages/rrweb/test/__snapshots__/integration.test.ts.snap:7239]: yeah, sucks a lot 😬

**Review by billyvg — COMMENTED** [review]
Should we add a test for dynamically inserted elements as well?

**Comment by mydea** [comment]
> Should we add a test for dynamically inserted elements as well?

I'll add some in the follow up PR!

**Pull request merged**